subprojects/docs/src/docs/userguide/troubleshooting/validation_problems.adoc

```diff
@@ -200,4 +200,13 @@ Because there are no semantics associated with this combination, Gradle cannot d
 
 To fix this, you most likely need to remove the conflicting modifier annotation or check that the actual property type is what you intended.
 
+[[incorrect_use_of_input_annotation]]
+== Incorrect use of the `@Input` annotation
+
+This error indicates that a property is annotated with `@Input`, but that it should be annotated with `@InputFile` or `@InputDirectory` instead.
+
+If you use the `@Input` annotation on a file-based property, Gradle wouldn't consider the file contents, or the directory contents, as inputs, as you might expect.
+
+To fix this problem, you need to tell Gradle if the file property represents an input file, in which case you should annotate it with `@InputFile`, or a directory, in which case it should be annotated with `@InputDirectory`.
```

Comment:
We should fix the Javadoc on `@Input`, `@InputFile` and `@InputDirectory` as well. See https://github.com/gradle/gradle/issues/5789